warehouse/query-core/src/main/java/datawave/query/language/parser/jexl/JexlNodeSet.java

```diff
@@ -12,26 +17,32 @@
 import java.util.Map;
 import java.util.Set;
 
+import static datawave.query.jexl.JexlASTHelper.nodeToKey;
+
 /**
  * Utility class that implements the {@link Set} interface for use with a collection of Jexl nodes.
  *
  * Element uniqueness is computed using {@link JexlASTHelper#nodeToKey(JexlNode)} for keys.
  *
- * Important Note: The set is considered invalid if any of the underlying Jexl nodes change.
+ * If a Jexl node and it's {@link ASTDelayedPredicate} equivalent are both added to the set, the set will accept the delayed predicate.
```

Comment:
In what cases will the set unwrap the marked node, and instead add the source node?  Are ASTDelayed nodes the only type which are handled this way?  Some more info here would be useful.